pkg/dockerfile/dockerfile.go

```diff
@@ -46,13 +51,26 @@ func Stages(opts *config.KanikoOptions) ([]config.KanikoStage, error) {
 		}
 		d, err = ioutil.ReadAll(response.Body)
 	} else {
+		lintDockerfilePath = opts.DockerfilePath
 		d, err = ioutil.ReadFile(opts.DockerfilePath)
 	}
 
 	if err != nil {
 		return nil, errors.Wrap(err, fmt.Sprintf("reading dockerfile at path %s", opts.DockerfilePath))
 	}
 
+	if opts.DockerfileLint {
+		if match {
+			if err := ioutil.WriteFile(lintDockerfilePath, d, 0644); err != nil {
```

Comment:
alternatively we could write `d` to a tempfile and run lint on the tmpfile. That would remove the need to declare the variable `lintDockerfilePath`. The benefit of this approach would be ensuring that you can always write the contents of the Dockerfile irrelevant of what kind of path is provided e.g. https:// s3:// or local file system, since in future maybe different methods will be added, but making sure lintDockerFilePath is set correctly might become more complicated, writing to a tempfile won't be.

Reply:
I guess here, the solution is to write `d` to `constants.DockerfilePath` only if the file doesn't exist yet.
Because in the end, it's exactly what is doing the function `copyDockerfile()` with as source `opts.DockerfilePath` which is not an URL. This way we can cover all the next possible cases without even having a variable `lintDockerfilePath`

Reply:
+1 that would be even cleaner